plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -226,18 +248,23 @@ func (s *SQLServer) Gather(acc telegraf.Accumulator) error {
 	}
 
 	var wg sync.WaitGroup
+	var mutex sync.Mutex
+	var healthMetrics = make(map[string]*HealthMetric)
 
 	for _, serv := range s.Servers {
 		for _, query := range s.queries {
 			wg.Add(1)
 			go func(serv string, query Query) {
 				defer wg.Done()
-				acc.AddError(s.gatherServer(serv, query, acc))
+				queryError := s.gatherServer(serv, query, acc)
+				s.gatherHealth(healthMetrics, serv, queryError, &mutex)
```

Comment:
How about
```suggestion
			if s.HealthMetric {
				mutex.Lock()
				s.gatherHealth(healthMetrics, serv, queryError)
				mutex.Unlock()
			}
```
so it's easier for people to see that this operation depends on the switch. You might want to do the same for `s.accHealth` but this keep the functions more simple. It's only a suggestion...

Reply:
I like it. Done.